carla/models/catalog/catalog.py

```diff
@@ -49,10 +47,14 @@ def __init__(
         super().__init__(data)
         self._backend = backend
 
+        # TODO: Add logic for feature_order depending on upcoming different catalog models
+        self._catalog = load_catalog(data.catalog_file, data.name)
         if self._backend == "pytorch":
             ext = "pt"
+            self._feature_input_order = self._catalog["feature_input_order_bin"]
         elif self._backend == "tensorflow":
             ext = "h5"
+            self._feature_input_order = self._catalog["feature_input_order_drop_first"]
```

Comment:
Will it always be that way that the tensorflow models be drop first and the other be bin? Isn't that just how we trained our models that we put on catalog?

Would it make sense to pass instead a flag how the model has been trained in the constructor and assign the feature order encording to the flag?

E.g. 
```
    def __init__(
        self,
        data,
        model_type,
        drop_first_encoded=True,
         ...,
        **kws
    ):
```
```
if drop_first_encoded=True:
    self._feature_order = data.feature_order_drop_first
else:
    self._feature_order = data.feature_order
```

(Also maybe we can simplify naming as e.g. above, if that makes sense?)

Reply:
I like this idea, but this would mean that a user would have to know which feature order is used by our catalog models.
For example, there would be no Pytorch model with drop_first.

I'd like to keep this logic away from the user, such that if a user wants to use a pre-trained model from our catalog, he only has to pass the DataCatalog to the MLModelCatalog and define which backend he wants.

Reply:
Ok also makes sense, I just don't understand why the way how we encode the data is dependend on the on the backend we use. Could we not also just train a Pytorch model without drop first?

Another idea, does it maybe make sense to introduce a configuration for our pretrained models, similar to the datacatalog.yaml?

Something like
```
  adult:
      ann:
          pytorch:
            encoding: "drop_first"
            feature_order: [
                ...
            ]
compas:
    linear:
        tensorflow:
            ....
```

Reply:
If we would only train a pytorch model without drop first, we could not use FACE because it only supports binary features with drop first, and CEM because it only works with Tensorflow sessions. 

You're right, I'll introduce a modelcatalog.yaml to handle all model specific configurations.

Reply:
Another suggestion, instead of passing the catalog file name to the constructors of DataCatalog and MLModelCatalog, could we keep this information away from the user and construct only two yaml-files data_catalog.yaml and mlmodel_catalog.yaml? Each file consists of information about every dataset, as you proposed above. Then a user would not have to know how we named our files.

I'd change this in the next commit, I think this would be a big improvement.